lib/turn.rb

```diff
@@ -102,27 +102,26 @@ def check_vertical_win
   end
 
   #range is acting as way to create index positions to check index positions on our board
-  def diag_offset_columns 
-    board = get_board_as_states
-    (1..3).map do |outer_int|
+  def diag_offset_columns(board, range)
+    range.map do |outer_int|
       (outer_int..(board[0].size - 1)).collect {|inner_int| board[inner_int - outer_int][inner_int]}
     end
   end
 
-  def diag_offset_rows
-    board = get_board_as_states
-    (0..2).map do |outer_int|
+  def diag_offset_rows(board, range)
+    range.map do |outer_int|
```

Comment:
Also as far as refactoring, the outer_int and inner_int naming might be worth renaming as it refers a bit too specifically to the looping, however, neither of us could think of something more logical so let's stick with it and readdress if/when we refactor.